File: VisualBasicAssignment/screen/kangtze/staffmenu.vb
Imports System.Data.SqlClient
Imports System.Drawing.Image
Imports System.Drawing.Imaging
Imports System.Drawing.Bitmap
Imports System.IO


Public Class staffmenu
    Private WithEvents pnl As Panel
    Private WithEvents pic As PictureBox
    Private WithEvents namelabel As Label
    Private WithEvents pricelabel As Label
    Private WithEvents desclabel As Label
    'Private WithEvents btnEdit As Button

    Private Sub custmenu_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ReLoad()
    End Sub

    Private Sub ReLoad()
        pnlPicture.Controls.Clear()
        Dim arrImage() As Byte
        Using db As FoodShopEntities1 = New FoodShopEntities1()
            Dim foods As List(Of Food) = DatabaseConnections.db.Foods.ToList()
            Try
                For i = 0 To foods.Count

                    arrImage = foods.Item(i).image
                    pic = New PictureBox
                    pic.Width = 130
                    pic.Height = 150
                    pic.BackgroundImageLayout = ImageLayout.Stretch

                    If arrImage IsNot Nothing Then
                        pic.BackgroundImage = ExFunctions.byteToImage(CType(arrImage, Byte()))
                    End If

                    namelabel = New Label
                    namelabel.Font = New Font("Garamond", 12.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
                    namelabel.Text = "Name : " + foods.Item(i).title
                    namelabel.Dock = DockStyle.Top

                    pricelabel = New Label
                    pricelabel.Font = New Font("Garamond", 12.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
                    pricelabel.Text = "Price : RM" + foods.Item(i).price.ToString
                    pricelabel.Dock = DockStyle.Top


                    desclabel = New Label
                    desclabel.Font = New Font("Garamond", 12.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
                    desclabel.Text = foods.Item(i).descriptions
                    desclabel.Dock = DockStyle.Fill

                    Dim btnEdit As New Button
                    btnEdit.Name = foods.Item(i).id

                    btnEdit.Text = "Edit"
                    AddHandler btnEdit.Click, AddressOf Me.Button_Click
                    btnEdit.Dock = DockStyle.Top

                    pnl = New Panel
                    pnl.Controls.Add(pricelabel)
                    pnl.Controls.Add(namelabel)
                    pnl.Controls.Add(desclabel)
                    pnl.Controls.Add(btnEdit)

                    pnlPicture.Controls.Add(pic)
                    pnlPicture.Controls.Add(pnl)

                Next
            Catch ex As Exception
                Console.WriteLine("Exception caught: {0}", ex)
            End Try
        End Using

    End Sub

    Private Sub Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Dim btn As New Button
        btn = CType(sender, Button)
        MessageBox.Show(btn.Name + "clicked")
        editid = btn.Name
        edit.ShowDialog()
        Me.Show()

    End Sub

    Private Sub btnReload_Click(sender As Object, e As EventArgs) Handles btnReload.Click
        ReLoad()
    End Sub
End Class

File: VisualBasicAssignment/screen/kangtze/staffmenu.cs
using System;
using System.Drawing;
using System.Linq;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace VisualBasicAssignment
{
    public partial class staffmenu
    {
        public staffmenu()
        {
            InitializeComponent();
            _btnReload.Name = "btnReload";
        }

        private Panel pnl;
        private PictureBox pic;
        private Label namelabel;
        private Label pricelabel;
        private Label desclabel;
        // Private WithEvents btnEdit As Button

        private void custmenu_Load(object sender, EventArgs e)
        {
            ReLoad();
        }

        private void ReLoad()
        {
            pnlPicture.Controls.Clear();
            byte[] arrImage;
            using (var db = new FoodShopEntities1())
            {
                var foods = DatabaseConnections.db.Foods.ToList();
                try
                {
                    for (int i = 0, loopTo = foods.Count; i <= loopTo; i++)
                    {
                        arrImage = foods[i].image;
                        pic = new PictureBox();
                        pic.Width = 130;
                        pic.Height = 150;
                        pic.BackgroundImageLayout = ImageLayout.Stretch;
                        if (arrImage is object)
                        {
                            pic.BackgroundImage = ExFunctions.byteToImage(arrImage);
                        }

                        namelabel = new Label();
                        namelabel.Font = new Font("Garamond", 12.0f, FontStyle.Regular, GraphicsUnit.Point, Conversions.ToByte(0));
                        namelabel.Text = "Name : " + foods[i].title;
                        namelabel.Dock = DockStyle.Top;
                        pricelabel = new Label();
                        pricelabel.Font = new Font("Garamond", 12.0f, FontStyle.Regular, GraphicsUnit.Point, Conversions.ToByte(0));
                        pricelabel.Text = "Price : RM" + foods[i].price.ToString();
                        pricelabel.Dock = DockStyle.Top;
                        desclabel = new Label();
                        desclabel.Font = new Font("Garamond", 12.0f, FontStyle.Regular, GraphicsUnit.Point, Conversions.ToByte(0));
                        desclabel.Text = foods[i].descriptions;
                        desclabel.Dock = DockStyle.Fill;
                        var btnEdit = new Button();
                        btnEdit.Name = foods[i].id.ToString();
                        btnEdit.Text = "Edit";
                        btnEdit.Click += Button_Click;
                        btnEdit.Dock = DockStyle.Top;
                        pnl = new Panel();
                        pnl.Controls.Add(pricelabel);
                        pnl.Controls.Add(namelabel);
                        pnl.Controls.Add(desclabel);
                        pnl.Controls.Add(btnEdit);
                        pnlPicture.Controls.Add(pic);
                        pnlPicture.Controls.Add(pnl);
                    }
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }
            }
        }

        private void Button_Click(object sender, EventArgs e)
        {
            var btn = new Button();
            btn = (Button)sender;
            MessageBox.Show(btn.Name + "clicked");
            Module1.editid = btn.Name;
            My.MyProject.Forms.edit.ShowDialog();
            Show();
        }

        private void btnReload_Click(object sender, EventArgs e)
        {
            ReLoad();
        }
    }
}